### ISSUE TITLE
RuntimeError: Cannot orchestrate task run. Failed to connect to API at http://127.0.0.1:4200/api/

### ISSUE THREAD

**Issue by ian-etnyre** [PrefectHQ/prefect#6759]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

when creating a flow that retrieves information such as JSON.load or other block load's the flow after some time the orion server stops responding and the exception appears

### Reproduction

```python
from prefect import flow, task, tags
import asyncio
import random
import os
from prefect.blocks.system import JSON
from prefect_dask import DaskTaskRunner


@task
async def do_work(idx: int, block_name: str):
    config = await JSON.load(block_name)
    sleep_time = random.uniform(.5, 3.0)
    await asyncio.sleep(sleep_time) # simulating load with random times
    print(f"done with {idx}")
    return idx


@flow(task_runner=DaskTaskRunner())
def concurrent(block_name: str):
    with tags("workers"):
        work_futures = [ do_work.submit(index, block_name) for index in list(range(1,1024)) ]
        work_results = [ item.result() for item in work_futures ]
    print("the end")


if __name__ == "__main__":
    os.system("prefect concurrency-limit create workers 4")
    json_block = JSON(value={"the_answer": 42})
    json_block.save("test-block", overwrite=True)
    concurrent("test-block")
```

### Error

```
18:26:49.101 | INFO    | Flow run 'rough-dove' - Created task run 'do_work-fc1fb0b6-146' for task 'do_work'
18:26:49.102 | INFO    | Flow run 'rough-dove' - Submitted task run 'do_work-fc1fb0b6-146' for execution.
18:26:49.186 | INFO    | Flow run 'rough-dove' - Created task run 'do_work-fc1fb0b6-392' for task 'do_work'
18:26:49.187 | INFO    | Flow run 'rough-dove' - Submitted task run 'do_work-fc1fb0b6-392' for execution.
18:26:49.339 | ERROR   | Flow run 'rough-dove' - Encountered exception during execution:
Traceback (most recent call last):
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_async/connection_pool.py", line 227, in handle_async_request
    connection = await status.wait_for_connection(timeout=timeout)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_async/connection_pool.py", line 34, in wait_for_connection
    await self._connection_acquired.wait(timeout=timeout)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_synchronization.py", line 38, in wait
    await self._event.wait()
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 1842, in wait
    if await self._event.wait():
  File "/usr/lib/python3.9/asyncio/locks.py", line 226, in wait
    await fut
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/client.py", line 377, in api_healthcheck
    await self._client.get("/health")
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1751, in get
    return await self.request(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1527, in request
    return await self.send(request, auth=auth, follow_redirects=follow_redirects)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/client.py", line 257, in send
    await super().send(*args, **kwargs)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1614, in send
    response = await self._send_handling_auth(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1642, in _send_handling_auth
    response = await self._send_handling_redirects(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1679, in _send_handling_redirects
    response = await self._send_single_request(request)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1716, in _send_single_request
    response = await transport.handle_async_request(request)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_transports/default.py", line 353, in handle_async_request
    resp = await self._pool.handle_async_request(req)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_async/connection_pool.py", line 232, in handle_async_request
    async with self._pool_lock:
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/client.py", line 378, in api_healthcheck
    return None
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/anyio/_core/_tasks.py", line 118, in __exit__
    raise TimeoutError
TimeoutError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/engine.py", line 595, in orchestrate_flow_run
    result = await run_sync(flow_call)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/utilities/asyncutils.py", line 57, in run_sync_in_worker_thread
    return await anyio.to_thread.run_sync(call, cancellable=True)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "/home/alburati/Proyectos/Aluxoft/orion/dask_concurent.py", line 22, in concurrent
    work_results = [ item.result() for item in work_futures ]
  File "/home/alburati/Proyectos/Aluxoft/orion/dask_concurent.py", line 22, in <listcomp>
    work_results = [ item.result() for item in work_futures ]
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/futures.py", line 225, in result
    return sync(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/utilities/asyncutils.py", line 240, in sync
    return run_async_from_worker_thread(__async_fn, *args, **kwargs)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/utilities/asyncutils.py", line 137, in run_async_from_worker_thread
    return anyio.from_thread.run(call)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/anyio/from_thread.py", line 49, in run
    return asynclib.run_async_from_thread(func, *args)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 970, in run_async_from_thread
    return f.result()
  File "/usr/lib/python3.9/concurrent/futures/_base.py", line 445, in result
    return self.__get_result()
  File "/usr/lib/python3.9/concurrent/futures/_base.py", line 390, in __get_result
    raise self._exception
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/futures.py", line 236, in _result
    return final_state.result(raise_on_failure=raise_on_failure)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/orion/schemas/states.py", line 143, in result
    raise data
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/task_runners.py", line 286, in _run_and_store_result
    self._results[key] = await call()
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/engine.py", line 1078, in begin_task_run
    raise RuntimeError(
RuntimeError: Cannot orchestrate task run '1ead6d09-96d5-489b-afc9-cdf02479cc5d'. Failed to connect to API at http://10.0.1.6:4200/api/.
18:26:49.427 | INFO    | Flow run 'rough-dove' - Created task run 'do_work-fc1fb0b6-52' for task 'do_work'
18:26:49.428 | INFO    | Flow run 'rough-dove' - Submitted task run 'do_work-fc1fb0b6-52' for execution.
18:26:49.565 | INFO    | Flow run 'rough-dove' - Created task run 'do_work-fc1fb0b6-23' for task 'do_work'
18:26:49.565 | INFO    | Flow run 'rough-dove' - Submitted task run 'do_work-fc1fb0b6-23' for execution.
...
...
18:27:08.952 | INFO    | Flow run 'rough-dove' - Created task run 'do_work-fc1fb0b6-153' for task 'do_work'
18:27:08.953 | INFO    | Flow run 'rough-dove' - Submitted task run 'do_work-fc1fb0b6-153' for execution.
18:27:09.111 | INFO    | Flow run 'rough-dove' - Created task run 'do_work-fc1fb0b6-453' for task 'do_work'
18:27:09.111 | INFO    | Flow run 'rough-dove' - Submitted task run 'do_work-fc1fb0b6-453' for execution.
18:27:09.200 | ERROR   | Flow run 'rough-dove' - Crash detected! Request to http://10.0.1.6:4200/api/task_runs/ failed.
Traceback (most recent call last):
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/backends/asyncio.py", line 33, in read
    return await self._stream.receive(max_bytes=max_bytes)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 1265, in receive
    await self._protocol.read_event.wait()
  File "/usr/lib/python3.9/asyncio/locks.py", line 226, in wait
    await fut
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_exceptions.py", line 8, in map_exceptions
    yield
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/backends/asyncio.py", line 35, in read
    return b""
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/anyio/_core/_tasks.py", line 118, in __exit__
    raise TimeoutError
TimeoutError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_transports/default.py", line 60, in map_httpcore_exceptions
    yield
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_transports/default.py", line 353, in handle_async_request
    resp = await self._pool.handle_async_request(req)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_async/connection_pool.py", line 253, in handle_async_request
    raise exc
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_async/connection_pool.py", line 237, in handle_async_request
    response = await connection.handle_async_request(request)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_async/connection.py", line 90, in handle_async_request
    return await self._connection.handle_async_request(request)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_async/http11.py", line 105, in handle_async_request
    raise exc
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_async/http11.py", line 84, in handle_async_request
    ) = await self._receive_response_headers(**kwargs)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_async/http11.py", line 148, in _receive_response_headers
    event = await self._receive_event(timeout=timeout)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_async/http11.py", line 177, in _receive_event
    data = await self._network_stream.read(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/backends/asyncio.py", line 35, in read
    return b""
  File "/usr/lib/python3.9/contextlib.py", line 135, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_exceptions.py", line 12, in map_exceptions
    raise to_exc(exc)
httpcore.ReadTimeout

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/engine.py", line 1321, in report_flow_run_crashes
    yield
  File "/usr/lib/python3.9/contextlib.py", line 634, in __aexit__
    cb_suppress = await cb(*exc_details)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 662, in __aexit__
    raise exceptions[0]
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/engine.py", line 926, in create_task_run_then_submit
    task_run = await create_task_run(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/engine.py", line 965, in create_task_run
    task_run = await flow_run_context.client.create_task_run(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/client.py", line 1692, in create_task_run
    response = await self._client.post(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1842, in post
    return await self.request(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1527, in request
    return await self.send(request, auth=auth, follow_redirects=follow_redirects)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/client.py", line 257, in send
    await super().send(*args, **kwargs)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1614, in send
    response = await self._send_handling_auth(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1642, in _send_handling_auth
    response = await self._send_handling_redirects(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1679, in _send_handling_redirects
    response = await self._send_single_request(request)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1716, in _send_single_request
    response = await transport.handle_async_request(request)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_transports/default.py", line 353, in handle_async_request
    resp = await self._pool.handle_async_request(req)
  File "/usr/lib/python3.9/contextlib.py", line 135, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_transports/default.py", line 77, in map_httpcore_exceptions
    raise mapped_exc(message) from exc
httpx.ReadTimeout

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_synchronization.py", line 38, in wait
    await self._event.wait()
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 1842, in wait
    if await self._event.wait():
  File "/usr/lib/python3.9/asyncio/locks.py", line 226, in wait
    await fut
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_exceptions.py", line 8, in map_exceptions
    yield
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_synchronization.py", line 38, in wait
    await self._event.wait()
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/anyio/_core/_tasks.py", line 118, in __exit__
    raise TimeoutError
TimeoutError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_transports/default.py", line 60, in map_httpcore_exceptions
    yield
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_transports/default.py", line 353, in handle_async_request
    resp = await self._pool.handle_async_request(req)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_async/connection_pool.py", line 234, in handle_async_request
    raise exc
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_async/connection_pool.py", line 227, in handle_async_request
    connection = await status.wait_for_connection(timeout=timeout)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_async/connection_pool.py", line 34, in wait_for_connection
    await self._connection_acquired.wait(timeout=timeout)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_synchronization.py", line 38, in wait
    await self._event.wait()
  File "/usr/lib/python3.9/contextlib.py", line 135, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_exceptions.py", line 12, in map_exceptions
    raise to_exc(exc)
httpcore.PoolTimeout

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/client.py", line 103, in with_injected_client
    return await fn(*args, **kwargs)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/engine.py", line 231, in create_then_begin_flow_run
    state = await begin_flow_run(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/engine.py", line 367, in begin_flow_run
    terminal_state = await orchestrate_flow_run(
  File "/usr/lib/python3.9/contextlib.py", line 651, in __aexit__
    raise exc_details[1]
  File "/usr/lib/python3.9/contextlib.py", line 634, in __aexit__
    cb_suppress = await cb(*exc_details)
  File "/usr/lib/python3.9/contextlib.py", line 193, in __aexit__
    await self.gen.athrow(typ, value, traceback)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/engine.py", line 1328, in report_flow_run_crashes
    await client.set_flow_run_state(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/client.py", line 1612, in set_flow_run_state
    response = await self._client.post(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1842, in post
    return await self.request(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1527, in request
    return await self.send(request, auth=auth, follow_redirects=follow_redirects)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/client.py", line 257, in send
    await super().send(*args, **kwargs)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1614, in send
    response = await self._send_handling_auth(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1642, in _send_handling_auth
    response = await self._send_handling_redirects(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1679, in _send_handling_redirects
    response = await self._send_single_request(request)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1716, in _send_single_request
    response = await transport.handle_async_request(request)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_transports/default.py", line 353, in handle_async_request
    resp = await self._pool.handle_async_request(req)
  File "/usr/lib/python3.9/contextlib.py", line 135, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_transports/default.py", line 77, in map_httpcore_exceptions
    raise mapped_exc(message) from exc
httpx.PoolTimeout

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/alburati/Proyectos/Aluxoft/orion/dask_concurent.py", line 30, in <module>
    concurrent("test-block")
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/flows.py", line 384, in __call__
    return enter_flow_run_engine_from_flow_call(
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/engine.py", line 158, in enter_flow_run_engine_from_flow_call
    return anyio.run(begin_run)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/anyio/_core/_eventloop.py", line 70, in run
    return asynclib.run(func, *args, **backend_options)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 292, in run
    return native_run(wrapper(), debug=debug)
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 287, in wrapper
    return await func(*args)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/client.py", line 103, in with_injected_client
    return await fn(*args, **kwargs)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/prefect/client.py", line 1929, in __aexit__
    return await self._exit_stack.__aexit__(*exc_info)
  File "/usr/lib/python3.9/contextlib.py", line 651, in __aexit__
    raise exc_details[1]
  File "/usr/lib/python3.9/contextlib.py", line 634, in __aexit__
    cb_suppress = await cb(*exc_details)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_client.py", line 1997, in __aexit__
    await self._transport.__aexit__(exc_type, exc_value, traceback)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpx/_transports/default.py", line 332, in __aexit__
    await self._pool.__aexit__(exc_type, exc_value, traceback)
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_async/connection_pool.py", line 326, in __aexit__
    await self.aclose()
  File "/home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages/httpcore/_async/connection_pool.py", line 312, in aclose
    raise RuntimeError(
RuntimeError: The connection pool was closed while 1020 HTTP requests/responses were still in-flight.
```

### Versions

```
Version:             2.3.2
API version:         0.8.0
Python version:      3.9.5
Git commit:          6e931ee9
Built:               Tue, Sep 6, 2022 12:36 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         hosted
```

### Additional context

The first time the example code is run it behaves differently than the following times that it is run.
On the first run, the example code will finish: tasks print before raising exceptions.
On the following runs, no prints happen before or after the exception.

**Comment by BitTheByte** [comment]
@ian-etnyre This is due the difference between your local environment and cluster environment 
in the cluster, there is no "http://127.0.0.1:4200" available prefect is using your local API address to communicate inside the cluster thus getting connection errors. can be simply fixed by pointing to your local API address to Orion service inside the cluster
```
export PREFECT_API_URL="http://orion:4200" 
```

keep in mind you'll need to have this entry in your hosts file locally and port forward 4200 from cluster
```
127.0.0.1     orion
````

This was my workaround, however, I believe that the agent should override `PREFECT_API_URL` to fix this problem

**Comment by ian-etnyre** [comment]
@discdiver the bug report

**Comment by madkinsz** [comment]
Looks related to or duplicate of https://github.com/PrefectHQ/prefect/issues/6425

**Comment by madkinsz** [comment]
Can you please share the output of `pip show httpx httpcore anyio` where this flow is running?

**Comment by ian-etnyre** [comment]
For the example code above I run `python dask_concurrent.py` having an orion server running as well on my local computer.

I created the example because I get this error in my production deployment that is running on dockers.

What I have noticed is lowering the number of tasks like `list(range(1, 24))` makes the error not come up.

here is `pip show` output:
```
Name: httpx
Version: 0.23.0
Summary: The next generation HTTP client.
Home-page: https://github.com/encode/httpx
Author: Tom Christie
Author-email: tom@tomchristie.com
License: BSD
Location: /home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages
Requires: certifi, httpcore, rfc3986, sniffio
Required-by: prefect
---
Name: httpcore
Version: 0.15.0
Summary: A minimal low-level HTTP client.
Home-page: https://github.com/encode/httpcore
Author: Tom Christie
Author-email: tom@tomchristie.com
License: BSD
Location: /home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages
Requires: anyio, certifi, sniffio, h11
Required-by: httpx
---
Name: anyio
Version: 3.6.1
Summary: High level compatibility layer for multiple asynchronous event loop implementations
Home-page: None
Author: Alex Grönholm
Author-email: alex.gronholm@nextday.fi
License: MIT
Location: /home/alburati/Proyectos/Aluxoft/repos/scicarta-pipeline/p39/lib/python3.9/site-packages
Requires: sniffio, idna
Required-by: starlette, prefect, httpcore

```

**Comment by madkinsz** [comment]
Thanks! Does this require use of the `DaskTaskRunner` to reproduce?

**Comment by jemos** [comment]
I suspect this is the same problem I'm having. If I launch a lot of tasks in parallel, prefect would crash. In my case increasing the healthcheck timeout from 10 to 60 seconds fixed the problem. From what I understood:

1. The healthcheck failed to connect due to some resource starvation (`orion.py / api_healthcheck()`)
2. The `api_healthcheck` function returns the exception
3. The agent that is running the flow throws a `RuntimeError`
4. Flow execution aborts (`RuntimeError`) and since there were some pending connections it shows this "The connection pool was closed while 1020 HTTP requests/responses were still in-flight." message and the "Crash detected! Execution was interrupted by an unexpected exception".

The patch I did was in `prefect/client/orion.py`, changing `fail_after` timeout from 10 to 60:

```
...
try:
            with anyio.fail_after(60):
                await self._client.get("/health")
                return None
...
```

(Or maybe better use `PREFECT_API_REQUEST_TIMEOUT` here also?)

It's possible something in the host could be configured to be capable of handling more connections?, but...

**Comment by ian-etnyre** [comment]
@madkinsz Sorry for the delay, yes this happens with out `DaskTaskRunner` as well.

**Comment by ian-etnyre** [comment]
Hi @jemos, I will test this out to see if it solves my problem too, Thanks

**Comment by madkinsz** [comment]
Hm the healthcheck could probably be updated to remove the `fail_after` and then it'd just respect the `PREFECT_API_REQUEST_TIMEOUT` setting.

**Comment by john-jam** [comment]
Hi! Adding some info if that could help.
Please let me know if you prefer that I fill a new issue.

## Bug summary
I also hit those kind of errors when running a `Dask KubeCluster` on a GCP k8s cluster with a big number of workers. It seems the `orion` API is saturated at some point.

The `orion` API was running on a `n1-standard-1` GCP machine (1 vCPU / 3.75GB RAM) and the error started to appear when using >20 workers. After I updated to `n1-standard-4` GCP machine (4 vCPU / 15 GB RAM), I was able to run >100 workers without problem.

@madkinsz Do you have any minimum/recommended system requirements for a self-hosted `orion` API ?

## Reproduction
```python
@task
def test_task() -> int:
    logger = get_run_logger()
    logger.info(f"Task started")
    time.sleep(1)
    logger.info(f"Task finished")


@flow(
    task_runner=DaskTaskRunner(
        cluster_class="dask_kubernetes.KubeCluster",
        cluster_kwargs={
            "pod_template": make_pod_spec(
                image="my-custom-image-with-python3.8",
                memory_request="1G",
            ),
            "deploy_mode": "local",
        },
        adapt_kwargs={"minimum": NUM_WORKERS, "maximum": NUM_WORKERS},
    )
)
def test_flow():
    for i in range(0, 2000):
        test_task.submit()


if __name__ == "__main__":
    test_flow()
```

## Orion API error
```
INFO:     10.182.42.15:59148 - "POST /task_runs/ HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 404, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/usr/local/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/fastapi/applications.py", line 270, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/cors.py", line 84, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 75, in __call__
    raise exc
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 64, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/usr/local/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 680, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 427, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/fastapi/applications.py", line 270, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 75, in __call__
    raise exc
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 64, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/usr/local/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 680, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 275, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 65, in app
    response = await func(request)
  File "/usr/local/lib/python3.10/site-packages/prefect/orion/utilities/server.py", line 101, in handle_response_scoped_depends
    response = await default_handler(request)
  File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 231, in app
    raw_response = await run_endpoint_function(
  File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 160, in run_endpoint_function
    return await dependant.call(**values)
  File "/usr/local/lib/python3.10/site-packages/prefect/orion/api/task_runs.py", line 49, in create_task_run
    model = await models.task_runs.create_task_run(
  File "/usr/local/lib/python3.10/site-packages/prefect/orion/database/dependencies.py", line 117, in async_wrapper
    return await fn(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/prefect/orion/models/task_runs.py", line 75, in create_task_run
    result = await session.execute(query)
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/ext/asyncio/session.py", line 215, in execute
    result = await greenlet_spawn(
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/util/_concurrency_py3k.py", line 126, in greenlet_spawn
    result = context.throw(*sys.exc_info())
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 1712, in execute
    result = conn._execute_20(statement, params or {}, execution_options)
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1705, in _execute_20
    return meth(self, args_10style, kwargs_10style, execution_options)
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 333, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1572, in _execute_clauseelement
    ret = self._execute_context(
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1943, in _execute_context
    self._handle_dbapi_exception(
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 2128, in _handle_dbapi_exception
    util.raise_(exc_info[1], with_traceback=exc_info[2])
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/util/compat.py", line 208, in raise_
    raise exception
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1900, in _execute_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 479, in execute
    self._adapt_connection.await_(
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/util/_concurrency_py3k.py", line 68, in await_only
    return current.driver.switch(awaitable)
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/util/_concurrency_py3k.py", line 121, in greenlet_spawn
    value = await result
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 454, in _prepare_and_execute
    self._handle_exception(error)
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 389, in _handle_exception
    self._adapt_connection._handle_exception(error)
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 684, in _handle_exception
    raise error
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 418, in _prepare_and_execute
    prepared_stmt, attributes = await adapt_connection._prepare(
  File "/usr/local/lib/python3.10/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 660, in _prepare
    prepared_stmt = await self._connection.prepare(operation)
  File "/usr/local/lib/python3.10/site-packages/asyncpg/connection.py", line 566, in prepare
    return await self._prepare(
  File "/usr/local/lib/python3.10/site-packages/asyncpg/connection.py", line 584, in _prepare
    stmt = await self._get_statement(
  File "/usr/local/lib/python3.10/site-packages/asyncpg/connection.py", line 413, in _get_statement
    types, intro_stmt = await self._introspect_types(
  File "/usr/local/lib/python3.10/site-packages/asyncpg/connection.py", line 459, in _introspect_types
    return await self.__execute(
  File "/usr/local/lib/python3.10/site-packages/asyncpg/connection.py", line 1684, in __execute
    return await self._do_execute(
  File "/usr/local/lib/python3.10/site-packages/asyncpg/connection.py", line 1734, in _do_execute
    result = await executor(stmt, timeout)
  File "asyncpg/protocol/protocol.pyx", line 182, in bind_execute
  File "asyncpg/protocol/protocol.pyx", line 696, in asyncpg.protocol.protocol.BaseProtocol._get_timeout_impl
asyncio.exceptions.TimeoutError
```

## Agent error / Flow run error
```
2022-10-10 03:00:40,987 - distributed.client - ERROR -
Traceback (most recent call last):
  File "/venvs/venv/lib/python3.8/site-packages/distributed/client.py", line 1579, in _close
    await self.scheduler_comm.close()
asyncio.exceptions.CancelledError
2022-10-10 03:00:40,991 - distributed.deploy.adaptive_core - INFO - Adaptive stop
03:00:41.023 | ERROR   | Flow run 'notorious-clam' - Crash detected! Execution was interrupted by an unexpected exception.
...
2022-10-10 03:00:42,137 - distributed.core - INFO - Starting established connection
2022-10-10 03:00:42,183 - distributed.scheduler - INFO - Remove worker <WorkerState 'tcp://10.182.34.201:40795', name: 51, status: running, memory: 0, processing: 0>
2022-10-10 03:00:42,183 - distributed.core - INFO - Removing comms to tcp://10.182.34.201:40795
03:00:42.185 | ERROR   | asyncio.events -
Traceback (most recent call last):
  File "/venvs/venv/lib/python3.8/site-packages/prefect/client/orion.py", line 82, in with_injected_client
    return await fn(*args, **kwargs)
  File "/venvs/venv/lib/python3.8/site-packages/prefect/engine.py", line 306, in retrieve_flow_then_begin_flow_run
    return await begin_flow_run(
  File "/venvs/venv/lib/python3.8/site-packages/prefect/engine.py", line 368, in begin_flow_run
    terminal_state = await orchestrate_flow_run(
  File "/usr/local/lib/python3.8/contextlib.py", line 679, in __aexit__
    raise exc_details[1]
  File "/usr/local/lib/python3.8/contextlib.py", line 189, in __aexit__
    await self.gen.athrow(typ, value, traceback)
  File "/venvs/venv/lib/python3.8/site-packages/prefect/engine.py", line 1370, in report_flow_run_crashes
    raise exc from None
  File "/venvs/venv/lib/python3.8/site-packages/prefect/engine.py", line 1350, in report_flow_run_crashes
    yield
  File "/usr/local/lib/python3.8/contextlib.py", line 662, in __aexit__
    cb_suppress = await cb(*exc_details)
  File "/venvs/venv/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 662, in __aexit__
    raise exceptions[0]
  File "/venvs/venv/lib/python3.8/site-packages/prefect/engine.py", line 940, in create_task_run_then_submit
    task_run = await create_task_run(
  File "/venvs/venv/lib/python3.8/site-packages/prefect/engine.py", line 985, in create_task_run
    task_run = await flow_run_context.client.create_task_run(
  File "/venvs/venv/lib/python3.8/site-packages/prefect/client/orion.py", line 1542, in create_task_run
    response = await self._client.post(
  File "/venvs/venv/lib/python3.8/site-packages/httpx/_client.py", line 1842, in post
    return await self.request(
  File "/venvs/venv/lib/python3.8/site-packages/httpx/_client.py", line 1527, in request
    return await self.send(request, auth=auth, follow_redirects=follow_redirects)
  File "/venvs/venv/lib/python3.8/site-packages/prefect/client/base.py", line 182, in send
    response.raise_for_status()
  File "/venvs/venv/lib/python3.8/site-packages/prefect/client/base.py", line 125, in raise_for_status
    raise PrefectHTTPStatusError.from_httpx_error(exc) from exc.__cause__
prefect.exceptions.PrefectHTTPStatusError: Server error '500 Internal Server Error' for url 'http://prefect-orion.prefect-v2.svc.cluster.local:4200/api/task_runs/'
Response: {'exception_message': 'Internal Server Error'}
For more information check: https://httpstatuses.com/500

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/venvs/venv/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 287, in wrapper
    return await func(*args)
  File "/venvs/venv/lib/python3.8/site-packages/prefect/client/orion.py", line 82, in with_injected_client
    return await fn(*args, **kwargs)
  File "/venvs/venv/lib/python3.8/site-packages/prefect/client/orion.py", line 1779, in __aexit__
    return await self._exit_stack.__aexit__(*exc_info)
  File "/usr/local/lib/python3.8/contextlib.py", line 679, in __aexit__
    raise exc_details[1]
  File "/usr/local/lib/python3.8/contextlib.py", line 662, in __aexit__
    cb_suppress = await cb(*exc_details)
  File "/venvs/venv/lib/python3.8/site-packages/httpx/_client.py", line 1997, in __aexit__
    await self._transport.__aexit__(exc_type, exc_value, traceback)
  File "/venvs/venv/lib/python3.8/site-packages/httpx/_transports/default.py", line 332, in __aexit__
    await self._pool.__aexit__(exc_type, exc_value, traceback)
  File "/venvs/venv/lib/python3.8/site-packages/httpcore/_async/connection_pool.py", line 326, in __aexit__
    await self.aclose()
  File "/venvs/venv/lib/python3.8/site-packages/httpcore/_async/connection_pool.py", line 312, in aclose
    raise RuntimeError(
RuntimeError: The connection pool was closed while 475 HTTP requests/responses were still in-flight.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/venvs/venv/lib/python3.8/site-packages/distributed/utils.py", line 742, in wrapper
    return await func(*args, **kwargs)
  File "/venvs/venv/lib/python3.8/site-packages/distributed/scheduler.py", line 4093, in add_worker
    await self.handle_worker(comm, address)
  File "/venvs/venv/lib/python3.8/site-packages/distributed/scheduler.py", line 5260, in handle_worker
    await self.handle_stream(comm=comm, extra={"worker": worker})
  File "/venvs/venv/lib/python3.8/site-packages/distributed/core.py", line 822, in handle_stream
    msgs = await comm.read()
  File "/venvs/venv/lib/python3.8/site-packages/distributed/comm/tcp.py", line 225, in read
    frames_nbytes = await stream.read_bytes(fmt_size)
asyncio.exceptions.CancelledError
```

## Versions
My `orion` api and `agent` are deployed with the help of the [prefect-helm repository](https://github.com/PrefectHQ/prefect-helm) version `2022.10.6`.

### Orion
```
Version:             2.5.0
API version:         0.8.2
Python version:      3.10.7
Git commit:          eac37918
Built:               Thu, Oct 6, 2022 12:41 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         hosted
```

### Flow
I am running the flow in a docker image with python 3.8.

**Comment by matthewbrookes** [comment]
I have been having the same issues and @jemos's solution works for me. Would you welcome a PR removing the `fail_after()` like @madkinsz suggests?

**Comment by madkinsz** [comment]
@john-jam 

> Do you have any minimum/recommended system requirements for a self-hosted orion API ?

I do not at this time. We are still working to optimize the server and client and haven't benchmarked performance on different hardware. We're hoping to get things to a point where the failure mode is a slowdown of your flows rather than a crash though. As @matthewbrookes noted, we have merged from improvements to the client in #7090, perhaps those will help.

@zangell44 does #7090 host our server with http2 support or will that require changes to our uvicorn invocation?

**Comment by ian-etnyre** [comment]
> EDIT: I see the change has been made and now just waiting for a release crossed_fingers
@matthewbrookes looking forward to the bug fix in release

> Do you have any minimum/recommended system requirements for a self-hosted orion API ?
@john-jam I'am running my test above in an 88 vcpu and 756GB memory the problem is not related to hardware.

**Comment by zangell44** [comment]
> @zangell44 does https://github.com/PrefectHQ/prefect/pull/7090 host our server with http2 support or will that require changes to our uvicorn invocation?

Uvicorn does not support http2 unfortunately. I believe you would have to modify the way you run the app to use hypercorn in order to use http2. Some discussion here https://github.com/encode/uvicorn/issues/47

**Comment by ian-etnyre** [comment]
Hi guys,
 I increased `PREFECT_API_REQUEST_TIMEOUT: 120` to make the timeout disappear
but now I get this error:
```
orion_agent-orion_server-1  |   File "/usr/local/lib/python3.9/site-packages/prefect/client.py", line 225, in raise_for_status
orion_agent-orion_server-1  |     raise PrefectHTTPStatusError.from_httpx_error(exc) from exc.__cause__
orion_agent-orion_server-1  | Exception: PrefectHTTPStatusError("Server error '500 Internal Server Error' for url 'http://192.168.100.112:4200/api/task_runs/d833a6af-40e9-4d35-af6e-fb9025db2d55/set_state'\nRespo
nse: {'exception_message': 'Internal Server Error'}\nFor more information check: https://httpstatuses.com/500")
```
Any other advice or work around ?

**Comment by jemos** [comment]
That looks like a 500 response from the server. Can you have a look at the server logs? Maybe it will give an hint.

**Comment by SubtleParesh** [comment]
This is still an issue with only 8 dask workers. While Prefect is running on 2 core/4gb ram with Postgress db. To note the number of tasks do exceed 300, and does work with ~200 tasks

**Comment by madkinsz** [comment]
@SubtleParesh can you please open a new issue with more details and a reproduction? A great deal has changed since this was opened.

**Comment by SubtleParesh** [comment]
Thanks. I just checked couple of more such issues and threads, going for retry of http error codes and decreasing number of concurrent tasks. Thanks a lot for a quick reply
